docs/development/modules/themes/creating-themes.md

```diff
@@ -0,0 +1,126 @@
+        Set("MyModule.MyManialink.Bg").To("000000");
+
+        return Task.CompletedTask;
+    }
+}
+```
+
+Here we set two options, one for the text color and one for the background color of our manialink. Note that we prefix these options with the module name and the manialink name. It is good practice to do it like this as it avoids conflicts with other themes. The option names also have certain naming conventions so that we can keep consistency between the option names.
+
+### Theme Option Naming Conventions
+In order to keep consistency and avoid conflicts and confusion, all theme options should follow these rules:
+
+- Default theme options should always be prefixed with `Default`.
+> *For example, let's say we have a default option called `MyModule.MyManialink.Text`. If this is the default option text color option for this manialink, it should be named `MyModule.MyManialink.Default.Text`. This allows us to separate the default options and additional options which may be optional.*
+
+- Always prefix the option name with the module name.
+> *This avoids conflicts with other modules as we can have multiple themes active at the same time.*
+
+- In addition to the module name, the option should be prefixed with the section of the Manialink, or the Component name itself.
+> *This avoid conflicts within the same module, and also helps understanding what the option is used for and where it belongs to. Note that in certain cases, such as having general purpose options for a module, one can omit this.*
+
+- Don't postfix options with redundant names such as "Color".
+> *Because these options are almost always used for colors, we can avoid verbose names such as "TextColor" or "BgColor", and simply call them "Text" and "Bg".*
+
+- Option names should be as simple and descriptive as possible, typically one word of it's primary usage.
+> *For example: "Text", "Bg" for BackGround, "Border", "Divider" etc.*
+
+- Postfix option names with their purpose if they have secondary usages, so that the name is PrimaryusageSecondaryusage.
+> *For example, let's say you have a secondary text color, you can call it TextSecondary. Other examples are BgHeaderGrad1, BgHeaderGrad2, BgContent, BgRow etc.*
+
+### Using Base Theme Options
+The constructor of a theme class allows Dependency Injection. This means that we can inject the Current Theme Object, use existing options for our theme. It allows us to use the base theme to style our Manialink, in order to keep a consistent style.
+
+You may be wondering, can't we just use the base theme options directly instead of creating a theme with new options that uses them? This is where the philosophy of the controller comes in. Because we wan't provide people with a highly customizable and versatile server controller so that they can make it exactly how they want. So everything within a Manialink has it's own option, which can be overriden in the config or by other Modules to modify the look of it.
```

Comment:
```suggestion
You may be wondering, can't we just use the base theme options directly instead of creating a theme with new options that uses them? This is where the philosophy of the controller comes in. We want to provide people with a highly customizable and versatile server controller so that they can make it exactly how they want. Everything within a Manialink has it's own options, which can be overridden in the config or by other modules to modify the look of it.
```